backend/endpoints/messages.py

```
@@ -77,3 +76,98 @@ async def send_message(
+                    "text": "check on updates"
+                }
+            }
+
+    Raises:
+        HTTPException [401]: You are not authorized to edit this message
+        HTTPException [404]: Message not found
+        HTTPException [424]: Message not edited
+        HTTPException [424]: Failure to retrieve data
+    """
+    DB = DataStorage(org_id)
+    message = await get_message(org_id, room_id, message_id)
+
+    try:
+        if not message:
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND, detail="Message not found"
+            )
+
+        payload = request.dict()
+        if message["sender_id"] != payload["sender_id"]:
+            raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail="You are not authorized to edit this message",
+            )
+
+        edit_message = await DB.update(
+            MESSAGE_COLLECTION, document_id=message_id, data=payload
+        )
+        if edit_message:
+            data = {
+                "room_id": room_id,
+                "message_id": message_id,
+                "sender_id": payload["sender_id"],
+                "text": payload["text"],
+            }
+            background_tasks.add_task(
+                centrifugo_client.publish, room_id, Events.MESSAGE_UPDATE, data
+            )  # publish to centrifugo in the background
+            return JSONResponse(
+                content=ResponseModel.success(data=data, message="message edited"),
+                status_code=status.HTTP_200_OK,
+            )
+        raise HTTPException(
+            status_code=status.HTTP_424_FAILED_DEPENDENCY,
+            detail={"message not edited": edit_message},
+        )
+    except HTTPException as e:
+        return JSONResponse(
+            data="Failure to retrieve data",
+            status=status.HTTP_424_FAILED_DEPENDENCY,
+            detail=e
+        )
```

Do we really need a try-except block here? If we do, then an `else` clause is needed because the `try` block should only contains codes that will throw expected errors.
Also the except Exception it a bit too broad. We should catch more concise exceptions.
But as it it now, most of the expected errors are handled in the code.
The other unexpected errors should fail loudly.

---

adjusted.